File: core/src/main/scala/com/graphbrain/db/Backend.scala
package com.graphbrain.db

import VertexType.VertexType

trait Backend {
	def close(): Unit
	def get(id: String, vtype: VertexType): Vertex
	def put(vertex: Vertex): Vertex
	def update(vertex: Vertex): Vertex
	def remove(vertex: Vertex)
	def associateEmailToUsername(email: String, username: String)
	def usernameByEmail(email: String): String
	def listByType(vtype: VertexType): List[Vertex]
	def edges(center: Vertex): Set[Edge]
  def edges(pattern: Edge): Set[Edge]
  def addLinkToGlobal(globalId: String, userId: String)
  def removeLinkToGlobal(globalId: String, userId: String)
  def alts(globalId: String): Set[String]
}

File: core/src/main/java/com/graphbrain/db/Backend.java
package com.graphbrain.db;

import java.util.List;
import java.util.Set;

interface Backend {
    public void close();
	public Vertex get(String id, VertexType vtype);
	public Vertex put(Vertex vertex);
    public Vertex update(Vertex vertex);
	public void remove(Vertex vertex);
	public void associateEmailToUsername(String email, String username);
	public String usernameByEmail(String email);
    public Set<Edge> edges(Edge pattern);
	public Set<Edge> edges(Vertex center);
    public void addLinkToGlobal(String globalId, String userId);
    public void removeLinkToGlobal(String globalId, String userId);
    public Set<String> alts(String globalId);
}